packages/flutter/lib/src/widgets/text.dart

```diff
@@ -210,16 +215,40 @@ class Text extends StatelessWidget {
     this.textScaleFactor,
     this.maxLines,
   }) : assert(data != null),
+       span = null,
        super(key: key);
 
+  /// Creates a text widget from a [TextSpan].
+  const Text.fromSpan(this.span, {
+    Key key,
+    this.textAlign,
+    this.textDirection,
+    this.softWrap,
+    this.overflow,
+    this.textScaleFactor,
+    this.maxLines,
+  }): assert(span != null),
+      data = null,
+      style = null,
+      super(key: key);
+
   /// The text to display.
+  ///
+  /// If constructed with the default constructor, `span` will be null.
   final String data;
 
+  /// The text to display as a span.
+  ///
+  /// If constructed from a [TextSpan], `data` will be null.
+  final TextSpan span;
```

Comment:
We should probably call this `textSpan` for clarity.

Reply:
Done